src/boot/main.c

```diff
@@ -333,6 +333,43 @@ void check_cache_emulation() {
     __osRestoreInt(saved);
 }
 
+/**
+ * Increment the first and last values of the stack.
+ * If they're different, that means an error has occured, so trigger a crash.
+*/
+#ifdef DEBUG
+void check_stack_validity(void) {
+    gIdleThreadStack[0]++;
+    gIdleThreadStack[THREAD1_STACK - 1]++;
+    if (gIdleThreadStack[0] != gIdleThreadStack[THREAD1_STACK - 1]) {
```

Comment:
Why not use the `assert` keyword here? That's kind of what it's for.

Reply:
assert didn't work when I first tried it

Reply:
:HOW:

Reply:
That to me just implies assert is actually broken then, which is frankly a very big problem. But at the same time I kinda doubt it's actually broken. Skill issue moment?

Reply:
yeah just a skill issue lol, asserts are only active with the debug flag, fixed that now the debug flag is required for this to work anyway